services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/extension/env/HardwareEnvCleanupExtension.groovy

```diff
@@ -0,0 +1,49 @@
+package org.openkilda.functionaltests.extension.env
+
+import org.openkilda.messaging.info.event.SwitchChangeType
+
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationContext
+
+/**
+ * This extension is responsible for cleaning the test environment up at the start of the test run
+ * against a hardware topology.<br>
+ * Turned on on-demand by passing a `-Dcleanup` property to the build params.<br>
+ * This has no guarantee to fully repair the broken env and should be used with care,
+ * especially if the target environment is shared between many people.<br>
+ *
+ * WARNING: do not over-use this option. Cleanup may be destructive and mask potential defects related to improper
+ * topology discovery, default rules setup on switch discovery, bandwidth discovery etc.
+ */
+@Slf4j
+class HardwareEnvCleanupExtension extends EnvCleanupExtension {
+    @Value('${cleanup:false}')
```

Comment:
I think it should name 'env.hardware.cleanup'. Also, it should be added to kilda.properties.example and 'confd'.

Reply:
> I think it should name 'env.hardware.cleanup'.

agree
> it should be added to kilda.properties.example and 'confd'

Maybe add to kilda.properties.example, but have doubts about 'confd'. I see this more as a flag which is being set when calling `mvn test...` command, rather that an option to be stored along with all the static properties. Not sure whether it belongs to the list of all kilda variables. My thought process is like: It is false by default, turn it on during the build if you know what you're doing.

Reply:
done, no confd